lib/hosting_environment.rb

```diff
@@ -4,15 +4,17 @@ module HostingEnvironment
     placements: "beta",
   }.with_indifferent_access.freeze
 
-  def self.name(current_service)
-    if Rails.env.production?
+  def self.env
+    @env ||= ActiveSupport::EnvironmentInquirer.new(ENV["HOSTING_ENV"].presence || "development")
+  end
+
+  def self.phase(current_service)
+    if env.production?
       PRODUCTION_BANNER_NAME.fetch(current_service)
+    elsif env.qa?
+      "QA"
     else
-      Rails.env
+      env
     end
   end
```

Comment:
This `HostingEnvironment.phase` method still looks & smells like a View concern to me.

Would it sit more comfortably in the `PhaseBannerHelper`?

Reply:
The values themselves are data, the "QA" string manipulation is a valid View concern though. I've moved that into a view helper.